plugins/logging/pkg/util/opensearch.go

```diff
@@ -28,7 +32,27 @@ func (c *AsyncOpensearchClient) WaitForInit() {
 	c.initCond.L.Unlock()
 }
 
-func (c *AsyncOpensearchClient) SetClient(setter func() *opensearch.Client) {
+func (c *AsyncOpensearchClient) WaitForInitWithTimeout(timeout time.Duration) bool {
+	c.initCond.L.Lock()
+	defer c.initCond.L.Unlock()
+
+	notifier := make(chan struct{})
+	go func() {
+		defer close(notifier)
+		for !c.initialized {
+			c.initCond.Wait()
+		}
+	}()
+
+	select {
+	case <-notifier:
+		return true
+	case <-time.After(timeout):
+		return false
```

Comment:
Need to make sure the goroutine above completes before returning, otherwise this can deadlock